routers/api/v1/repo/collaborators.go

```diff
@@ -11,8 +11,72 @@ import (
 	"code.gitea.io/gitea/modules/context"
 )
 
+// ListCollaborators list a repository's collaborators
+func ListCollaborators(ctx *context.APIContext) {
+	access, err := models.AccessLevel(ctx.User, ctx.Repo.Repository)
+	if err != nil {
+		ctx.Error(500, "AccessLevel", err)
+		return
+	}
+	if access < models.AccessModeWrite {
```

Comment:
This should only be checked for private repos. For public repos anyone can fetch the list of collaborators :)